marklogic-data-hub/src/main/resources/ml-modules/root/com.marklogic.smart-mastering/algorithms/base.xqy

```diff
@@ -42,30 +42,39 @@ declare variable $_cached-algorithms-map as map:map := map:map();
  : @param $algorithms-xml the algorithms element from the matcher options
  : @return a map:map of the match algorithm functions
  :)
-declare function algorithms:build-algorithms-map($algorithms-xml as element(matcher:algorithms))
+declare function algorithms:build-algorithms-map($algorithms as node()*)
```

Comment:
This is an adjustment to accept either XML or JSON. Since this used by the compiled options, it will need to accept multiple versions of the options.